docs/src/pages/getting-started/expo.mdx

```diff
@@ -0,0 +1,200 @@
+
+      // !!! Whatever is returned here is sent to the clientside `onClientUploadComplete` callback
+      return { uploadedBy: metadata.userId };
+    }),
+} satisfies FileRouter;
+export type UploadRouter = typeof uploadRouter;
+```
+
+### Create an API route using the FileRouter
+
+```ts copy filename="src/app/api/uploadthing+api.ts"
+import { createRouteHandler } from "uploadthing/server";
+
+const uploadRouter = { ... } satisfies FileRouter;
+export type UploadRouter = typeof uploadRouter;
+
+export const { GET, POST } = createRouteHandler({
+  router: uploadRouter,
+
+  // Apply an (optional) custom config:
+  // config: { ... },
+});
+```
+
+> See configuration options in
+> [server API reference](/api-reference/server#createroutehandler)
+
+### Generate typed hooks
+
+Unlike the other UploadThing packages, the Expo package does not include any
+prebuilt components. Instead, we provide some helper hooks to help interact with
+the native file pickers.
+
+```tsx copy filename="src/utils/uploadthing.ts"
+import { generateReactNativeHelpers } from "@uploadthing/expo";
+
+import type { UploadRouter } from "~/app/api/uploadthing+api";
+
+export const { useImageUploader, useDocumentUploader } =
+  generateReactNativeHelpers<UploadRouter>({
+    /**
+     * Your server url.
+     * @default process.env.EXPO_PUBLIC_SERVER_URL
+     * @remarks In dev we will also try to use Expo.debuggerHost
+     */
+    url: "https://my-server.com",
+  });
+```
+
+### Use the FileRouter in your app
+
+```tsx copy filename="src/routes/index.tsx"
+import { openSettings } from "expo-linking";
+import { Alert, Pressable, StyleSheet, Text, View } from "react-native";
+
+import { useImageUploader } from "~/utils/uploadthing";
+
+export default function Home() {
+  const { openImagePicker, isUploading } = useImageUploader({
+    /**
+     * Any props here are forwarded to the underlying `useUploadThing` hook.
+     * Refer to the React API reference for more info.
+     */
+    onClientUploadComplete: () => Alert.alert("Upload Completed"),
+    onUploadError: (error) => Alert.alert("Upload Error", error.message),
+  });
+
+  return (
+    <View>
+      <Pressable
+        style={styles.button}
+        onPress={() => {
+          openImagePicker({
+            input, // Matches the input schema from the FileRouter endpoint
+            source: "library", // or "camera"
+            onInsufficientPermissions: () => {
+              Alert.alert(
+                "No Permissions",
+                "You need to grant permission to your Photos to use this",
+                [
+                  { text: "Dismiss" },
+                  { text: "Open Settings", onPress: openSettings },
+                ],
+              );
+            },
+          })
+        }}
+      >
+        <Text>Select Image</Text>
+      </Pressable>
+    </View>
+  );
+}
+
+const styles = StyleSheet.create({
+  button: { ... },
+});
+```
+
+</Steps>
+
+## Deployment
+
+TODO
```

Comment:
```suggestion
Follow the [Expo Router API Routes Deployment Guide](https://docs.expo.dev/router/reference/api-routes/#deployment)
```

Reply:
In the future we can include info on hosting server elsewhere with like Express but I think that's too nuanced for a "getting started" page